shcDecoder/src/main/java/ca/bc/gov/shcdecoder/SHCVerifierImpl.kt

```diff
@@ -105,28 +108,36 @@ class SHCVerifierImpl(
                     vaxCode?.toInt() == vaccineRule.cvxCode
                 }
 
-                when (ruleSet?.type) {
-                    1 -> {
-                        mrnType += ruleSet.ru
-                    }
+                val hasNumberOfDasElapsedSinceLastDose = intervalPassed(lastVaxDate, minInterval)
 
-                    2 -> {
-                        nrvvType += ruleSet.ru
-                    }
+                if (hasNumberOfDasElapsedSinceLastDose) {
+                    when (ruleSet?.type) {
+                        1 -> {
+                            mrnType += ruleSet.ru
+                        }
+
+                        2 -> {
+                            nrvvType += ruleSet.ru
+                        }
 
-                    3 -> {
-                        winacType += ruleSet.ru
+                        3 -> {
+                            winacType += ruleSet.ru
+                        }
                     }
                 }
-
                 val vaxDate = entry.resource.occurrenceDateTime?.toDate()
+                lastVaxDate = vaxDate
+                minInterval = ruleSet?.minDays ?: 0
                 val enoughDoses = mrnType >= rule.ruRequired
                     || nrvvType >= rule.ruRequired
                     || winacType >= rule.ruRequired
                 val enoughMixedDoses = rule.mixTypesAllowed
                     && (mrnType + nrvvType + winacType >= rule.mixTypesRuRequired)
                 if (enoughDoses || enoughMixedDoses) {
-                    return if (intervalPassed(vaxDate, rule)) {
+                    return if (rule.intervalRequired && intervalPassed(
```

Comment:
I think this will always return PARTIALLY if `!rule.intervalRequired`

Reply:
Yes, I just missed the condition while refactoring. let me fix it